File: Structural/Adapter/python/Reverse.py
class Target:
  def getMessage(self) -> str:
    return 'My first message'

class DifferentTarget:
  def getDifferentMessage(self) -> str:
    return 'egassem desreveR'

class TargetAdapter:
  def __init__(self, adaptee: DifferentTarget):
    self.adaptee = adaptee
    pass

  def getMessage(self):
    message = self.adaptee.getDifferentMessage()
    reverse = list(message)
    return ''.join(reverse[::-1])

File: Structural/Adapter/src/Reverse.ts
import { ITarget, IDifferentTarget } from './inderfaces'

class Target implements ITarget {
  getMessage(): string {
    return 'My target message'
  }
}

class DifferentTarget implements IDifferentTarget {
  getDifferentMessage(): string {
    return 'egassem desreveR'
  }
}

class TargetAdapter implements ITarget {
  constructor(
    private adaptee: IDifferentTarget
  ) {}

  getMessage(): string {
    return this.adaptee.getDifferentMessage()
      .split('')
      .reverse()
      .join('')
  }
}

export {
  Target,
  TargetAdapter,
  DifferentTarget
}